packages/nextjs/app/builders/[address]/_components/UserProfileCard.tsx

```diff
@@ -2,15 +2,19 @@
 
 import { UserLocation } from "./UserLocation";
 import { UserSocials } from "./UserSocials";
+import { useAccount } from "wagmi";
 import EditIcon from "~~/app/_assets/icons/EditIcon";
 import { UPDATE_USER_MODAL_ID, UpdateUserModal } from "~~/app/_components/UpdateUserModal";
 import { PunkBlockie } from "~~/components/PunkBlockie";
 import { Address } from "~~/components/scaffold-eth/Address/Address";
-import { useAuthSession } from "~~/hooks/useAuthSession";
+import { useUser } from "~~/hooks/useUser";
+import { UserRole } from "~~/services/database/config/types";
 import { UserByAddress } from "~~/services/database/repositories/users";
 
 export const UserProfileCard = ({ user, address }: { user: NonNullable<UserByAddress>; address: string }) => {
-  const { isAdmin } = useAuthSession();
+  const { address: connectedAddress } = useAccount();
+  const { data: connectedUser } = useUser(connectedAddress);
```

Comment:
Umm instead of using `useUser` maybe we could use `useAuthSession`   for `isAdmin` ? 

Kindof nit but this will also make sure people won't able to use impersonator

Reply:
I didn't even try impersonator previously, so didn't know how it works, interesting! So the reason for using `useAuthSession` here is to hide Admin buttons for the ones who use impersonator, right? So, we need to hide all other buttons in rainbow wallet menu before Siwe. So maybe to do Siwe automatically when user with role Admin is logged in? So we won't need indicators and redirects we can avoid https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/166#discussion_r2082099436

Reply:
Thanks for the answers!

Used `useAuthSession` [9ed1058](https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/166/commits/9ed1058dc1889a20b39e423aad5554d77044ac3a)

Closed #167 for now